libOTe/Tools/SubfieldPprf.h

```diff
@@ -0,0 +1,1437 @@
+            auto curSize = std::min<u64>(totalTrees - tIdx, 8);
+            if (curSize == 8)
+            {
+
+                for (u64 i = 0; i < output.rows(); ++i)
+                {
+                    auto oi = output[i].subspan(tIdx, 8);
+                    auto& ii = lvl[i];
+                    oi[0] = ii[0];
+                    oi[1] = ii[1];
+                    oi[2] = ii[2];
+                    oi[3] = ii[3];
+                    oi[4] = ii[4];
+                    oi[5] = ii[5];
+                    oi[6] = ii[6];
+                    oi[7] = ii[7];
+                }
+            }
+            else
+            {
+                for (u64 i = 0; i < output.rows(); ++i)
+                {
+                    auto oi = output[i].subspan(tIdx, curSize);
+                    auto& ii = lvl[i];
+                    for (u64 j = 0; j < curSize; ++j)
+                        oi[j] = ii[j];
+                }
+            }
+
+        }
+        else if (oFormat == PprfOutputFormat::BlockTransposed)
+        {
+
+            auto curSize = std::min<u64>(totalTrees - tIdx, 8);
+            if (curSize == 8)
+            {
+                for (u64 i = 0; i < output.cols(); ++i)
+                {
+                    auto& ii = lvl[i];
+                    output(tIdx + 0, i) = ii[0];
+                    output(tIdx + 1, i) = ii[1];
+                    output(tIdx + 2, i) = ii[2];
+                    output(tIdx + 3, i) = ii[3];
+                    output(tIdx + 4, i) = ii[4];
+                    output(tIdx + 5, i) = ii[5];
+                    output(tIdx + 6, i) = ii[6];
+                    output(tIdx + 7, i) = ii[7];
+                }
+            }
+            else
+            {
+                for (u64 i = 0; i < output.cols(); ++i)
+                {
+                    auto& ii = lvl[i];
+                    for (u64 j = 0; j < curSize; ++j)
+                        output(tIdx + j, i) = ii[j];
+                }
+            }
+
+        }
+        else if (oFormat == PprfOutputFormat::Interleaved)
+        {
+            std::cout << "no op" << std::endl;
+            // no op
+        }
+        else if (oFormat == PprfOutputFormat::Callback)
+            callback(tIdx, lvl);
+        else
+            throw RTE_LOC;
+    }
+    u64 interleavedPoint(u64 point, u64 treeIdx, u64 totalTrees, u64 domain, PprfOutputFormat format);
+    void interleavedPoints(span<u64> points, u64 domain, PprfOutputFormat format);
+    u64 getActivePath(const span<u8>& choiceBits);
+
+    template<typename G, typename F>
+    class SilentSubfieldPprfSender : public TimerAdapter
+    {
+    public:
+        u64 mDomain = 0, mDepth = 0, mPntCount = 0;
+        std::vector<G> mValue;
+        bool mPrint = false;
+        TreeAllocator mTreeAlloc;
+        Matrix<std::array<block, 2>> mBaseOTs;
+        
+        std::function<void(u64 treeIdx, span<AlignedArray<F, 8>>)> mOutputFn;
+        std::function<F (const block& b)> fromBlock;
```

Comment:
this will have very bad performance. Its ok for now but at some point we will need to switch to having the caller pass in a `type trait` object that defines `F,G`, and all of these related functions. That way these function calls can be inlined.